subprojects/core/src/main/java/org/gradle/api/internal/TaskOutputsInternal.java

```diff
@@ -47,9 +49,8 @@
     boolean hasDeclaredOutputs();
 
     /**
-     * Returns {@code false} if the task declares any multiple-output properties via {@link #files(Object...)},
-     * {@literal @}{@link org.gradle.api.tasks.OutputFiles} or
-     * {@literal @}{@link org.gradle.api.tasks.OutputDirectories}; or {@code true} otherwise.
+     * Returns whether the outputs can be cached.
+     * @param reasons Reasons for disabling the cache are added to this list.
      */
-    boolean isCacheAllowed();
+    boolean isCacheEnabled(List<TaskCachingDisabledReason> reasons);
```

Comment:
Looking at how this is used... do we have any strong reason to make the reasons something other than a String?